### PR TITLE
Fix incorrect code generated with const

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/31394
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/31867

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by HindujaB — APPROVED** [review]
LGTM

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/bytetype/BByteValueTest.java [hunk 1]
```diff
@@ -564,7 +564,15 @@ public void testBitwiseUnsignedRightShiftOperator2() {
         invokeUnsignedRightShiftOperatorTestFunction(a, i, j);
     }
 
-    private void invokeUnsignedRightShiftOperatorTestFunction(byte a, long i, long j) {
+    @Test(description = "Test bitwise unsigned right shift operator 2")
+    public void testBitwiseUnsignedRightShiftOperator3() {
+        int a = 0xff;
+        long i = -23445834;
+        long j = 5;
+        invokeUnsignedRightShiftOperatorTestFunction(a, i, j);
```
Comment (riyafa) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/bytetype/BByteValueTest.java:572]: shall we test when it is a `const` as in the issue?

Comment (riyafa) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/bytetype/BByteValueTest.java:572]: Shall we also test what happens when it is a byte shifted by a large value? We need to see if it becomes zero.

Comment (riyafa) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/bytetype/BByteValueTest.java:572]: Shall we test the left shift cases also?

Comment (warunalakshitha) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/bytetype/BByteValueTest.java:572]: Added test for byte shift with large value and becomes 0. Left shift cases are already added through PR. 

For constants, IMO since constants are compile-time, the constant operation should be work the same as the underlying type operation at runtime. Any issue with constant operation should be reproducible by a given type of operation. Adding a test would be redundant for me.

Comment (riyafa) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/bytetype/BByteValueTest.java:572]: The issue says `In both cases changing const to int makes the problem go away.` That's why I thought it might be an issue with the `const`. Don't you think it would be better to have atleast the test from the issue https://github.com/ballerina-platform/ballerina-lang/issues/31394

**Review by riyafa — APPROVED** [review]
